src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,239 @@
+                        
+                    if (string.IsNullOrEmpty(targetPlatformAlias))
+                    {
+                        return targetFrameworkAlias;
+                    }
+                        
+                    return targetFrameworkAlias + "-" + targetPlatformAlias + complexTargetFramework.TargetPlatformVersion;
+                }
+
+                return targetFrameworkAlias;
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Retrieves the target framework alias (i.e. net5.0) from the project's subscription service.
+        /// </summary>
+        /// <returns></returns>
+        private async Task<string> GetTargetFrameworkAlias(string targetFrameworkMoniker)
+        {
+            IProjectSubscriptionService? subscriptionService = _configuredProject.Services.ProjectSubscription;
+            Assumes.Present(subscriptionService);
+
+            IImmutableDictionary<string, IProjectRuleSnapshot> supportedTargetFrameworks = await subscriptionService.ProjectRuleSource.GetLatestVersionAsync(_configuredProject, new string[] { SupportedTargetFramework.SchemaName });
+            IProjectRuleSnapshot targetFrameworkRuleSnapshot = supportedTargetFrameworks[SupportedTargetFramework.SchemaName];
+
+            IImmutableDictionary<string, string>? targetFrameworkProperties = targetFrameworkRuleSnapshot.GetProjectItemProperties(targetFrameworkMoniker);
+            
+            if (targetFrameworkProperties != null &&
+                targetFrameworkProperties.TryGetValue(SupportedTargetFramework.AliasProperty, out string? targetFrameworkAlias))
+            {
+                return targetFrameworkAlias;
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Retrieves the associated platform alias (i.e. windows) from the project's subscription service.
+        /// </summary>
+        /// <param name="targetPlatformIdentifier"></param>
+        /// <returns></returns>
+        private async Task<string> GetTargetPlatformAlias(string targetPlatformIdentifier)
+        {
+            IProjectSubscriptionService? subscriptionService = _configuredProject.Services.ProjectSubscription;
+            Assumes.Present(subscriptionService);
+
+            IImmutableDictionary<string, IProjectRuleSnapshot> sdkSupportedTargetPlatformIdentifiers = await subscriptionService.ProjectRuleSource.GetLatestVersionAsync(_configuredProject, new string[] { SdkSupportedTargetPlatformIdentifier.SchemaName });
+
+            IProjectRuleSnapshot sdkSupportedTargetPlatformIdentifierRuleSnapshot = sdkSupportedTargetPlatformIdentifiers[SdkSupportedTargetPlatformIdentifier.SchemaName];
+
+            // The SdkSupportedTargetPlatformIdentifier rule stores the alias in the key value.
+            if (sdkSupportedTargetPlatformIdentifierRuleSnapshot.Items.TryGetKey(targetPlatformIdentifier, out string targetPlatformAlias))
```

Comment:
`TryGetKey` looks suspicious here. The first argument is a key. So this means "find the dictionary entry that has key `targetPlatformIdentifier` and return that key's value as `targetPlatformAlias`". It doesn't make sense to mix TPI and TPA.

I think you need something more like:

```c#
string targetPlatformAlias = sdkSupportedTargetPlatformIdentifierRuleSnapshot.Items
    .FirstOrDefault(entry => StringComparers.PropertyValues.Equals(entry.Value, targetPlatformIdentifier))
    ?.Key;
```

Reply:
This is a tricky one, the structure ends up looking like this:
![image](https://user-images.githubusercontent.com/8518253/118905600-e1723a00-b8d0-11eb-9b97-d7a228cf52f3.png)
The `RuleSnapshot` has a list of Items, which the first element contains the key-value: "windows", and the list of other properties like DisplayName ("Windows"), ProjectFullPath and ProjectDirectory.

I tried with the code you suggested and when asked to get the TPA for 'windows' it returned an empty value. I'll come back to further investigate this one.